### PR TITLE
feat(map_loader): add selected map loader

### PR DESCRIPTION
**Pull request by yukkysaito**
## Description

These PRs must be merged before these PRs.
- [PR for autoware_msg](https://github.com/autowarefoundation/autoware_msgs/pull/57)
   - add srv and msg for selected map loader
- [PR for autoware_launch](https://github.com/autowarefoundation/autoware_launch/pull/285)
   - modify config for `pointcloud_map_loader`

I would like to add selected map loader and metadata publisher to `pointcloud_map_loader`.

See https://github.com/orgs/autowarefoundation/discussions/2812#discussioncomment-5540316 for detail.

## Related links

- [TIER IV COMPANY INTERNAL LINK](https://tier4.atlassian.net/browse/T4PB-23166)
- [PR for autoware_msg](https://github.com/autowarefoundation/autoware_msgs/pull/57)
   - add srv and msg for selected map loader
- [PR for autoware_launch](https://github.com/autowarefoundation/autoware_launch/pull/285)
   - modify config for `pointcloud_map_loader`

## Tests performed

I tested with a data from Autoware tutorial. The map is divided into 20m grids ([sample-map-rosbag_split.zip](https://github.com/autowarefoundation/autoware.universe/files/10349104/sample-map-rosbag_split.zip)).

I have confirmed that the Autoware performs the same as the current Autoware when `enable_selected_load` is set false.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: map/map_loader/src/pointcloud_map_loader/selected_map_loader_module.cpp [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+  rclcpp::QoS durable_qos{1};
+  durable_qos.transient_local();
+  pub_metadata_ = node->create_publisher<autoware_map_msgs::msg::PointCloudMapMetaData>(
+    "output/pointcloud_map_metadata", durable_qos);
+  pub_metadata_->publish(createMetadata(all_pcd_file_metadata_dict_));
+}
+
+bool SelectedMapLoaderModule::onServiceGetSelectedPointCloudMap(
+  GetSelectedPointCloudMap::Request::SharedPtr req,
+  GetSelectedPointCloudMap::Response::SharedPtr res)
```
Comment (kminoda) [inline map/map_loader/src/pointcloud_map_loader/selected_map_loader_module.cpp:65]: Please make this a `const` function

**Review by kminoda — COMMENTED** [review]
Thank you! Mostly seems good

File: map/map_loader/src/pointcloud_map_loader/selected_map_loader_module.cpp (hunk 1 above)
Comment (Shin-kyoto) [reply map/map_loader/src/pointcloud_map_loader/selected_map_loader_module.cpp]: Thanks!
I added const in https://github.com/autowarefoundation/autoware.universe/pull/3286/commits/3b2d764a15d782ac5b822e27193c621832f4b1ba

And we need to change [onServiceGetPartialPointCloudMap](https://github.com/autowarefoundation/autoware.universe/blob/6b55253b86bdb943f5d9ee865919d679c05503aa/map/map_loader/src/pointcloud_map_loader/partial_map_loader_module.cpp#L53-L60) and [onServiceGetDifferentialPointCloudMa](https://github.com/autowarefoundation/autoware.universe/blob/6b55253b86bdb943f5d9ee865919d679c05503aa/map/map_loader/src/pointcloud_map_loader/differential_map_loader_module.cpp#L66-L75) into const in another PR.

**Comment by Shin-kyoto** [comment]
@kminoda
Now I finished to change corresponding to @kminoda's review. Can you submit your review?

**Comment by yukkysaito** [comment]
@Shin-kyoto Thank you for your support :+1:

**Review by kminoda — APPROVED** [review]
LGTM

**Pull request merged**